packages/ripple-keypairs/src/index.ts

```diff
@@ -38,66 +45,79 @@ const secp256k1 = {
     privateKey: string
     publicKey: string
   } {
-    const prefix = '00'
-
+    const derived = derivePrivateKey(entropy, options)
     const privateKey =
-      prefix + derivePrivateKey(entropy, options).toString(16, 64).toUpperCase()
+      SECP256K1_PREFIX + bytesToHex(numberToBytesBE(derived, 32))
 
-    const publicKey = bytesToHex(
-      Secp256k1.keyFromPrivate(privateKey.slice(2))
-        .getPublic()
-        .encodeCompressed(),
-    )
+    const publicKey = bytesToHex(nobleSecp256k1.getPublicKey(derived, true))
     return { privateKey, publicKey }
   },
 
-  sign(message, privateKey): string {
-    return bytesToHex(
-      Secp256k1.sign(hash(message), hexToBytes(privateKey), {
-        canonical: true,
-      }).toDER(),
+  sign(message: ByteArray, privateKey: HexString): string {
+    // Some callers pass the privateKey with the prefix, others without.
+    // elliptic.js implementation ignored the prefix, interpreting it as a
+    // leading zero byte. @noble/curves will throw if the key is not exactly
+    // 32 bytes, so we normalize it before passing to the sign method.
+    // TODO: keep back compat like this, or simply always require prefix as
+    // the ed25519 sign method does.
+    assert.ok(
+      (privateKey.length === 66 && privateKey.startsWith(SECP256K1_PREFIX)) ||
+        privateKey.length === 64,
     )
+    const normed = privateKey.length === 66 ? privateKey.slice(2) : privateKey
+    return nobleSecp256k1
+      .sign(hash(message), normed)
+      .toDERHex(true)
+      .toUpperCase()
   },
 
   verify(message, signature, publicKey): boolean {
-    return Secp256k1.verify(hash(message), signature, hexToBytes(publicKey))
+    const decoded = nobleSecp256k1.Signature.fromDER(signature)
+    return nobleSecp256k1.verify(decoded, hash(message), publicKey)
   },
 }
 
 const ed25519 = {
-  deriveKeypair(entropy: Uint8Array): {
+  deriveKeypair(entropy: ByteArray): {
     privateKey: string
     publicKey: string
   } {
     const prefix = 'ED'
     const rawPrivateKey = hash(entropy)
     const privateKey = prefix + bytesToHex(rawPrivateKey)
     const publicKey =
-      prefix + bytesToHex(Ed25519.keyFromSecret(rawPrivateKey).pubBytes())
+      prefix + bytesToHex(nobleEd25519.getPublicKey(rawPrivateKey))
     return { privateKey, publicKey }
   },
 
-  sign(message, privateKey): string {
-    // caution: Ed25519.sign interprets all strings as hex, stripping
-    // any non-hex characters without warning
-    if (!Array.isArray(message)) {
-      throw new Error('message must be array of octets')
-    }
+  sign(message: ByteArray, privateKey: HexString): string {
+    assert.ok(
+      Array.isArray(message) || message instanceof Uint8Array,
```

Comment:
TODO:
Could allow hex strings here now

Reply:
It actually probably doesn't matter given ripple-keypairs exports sign/verify which only works with hex strings ...